datasploit.py

```diff
@@ -45,45 +46,64 @@ def main(argv):
         shutil.copyfile(config_sample_path,config_file_path)
         print "[+] A config file is added please follow guide at https://datasploit.github.io/datasploit/apiGeneration/ to fill API Keys for better results"
         # We can think about quiting at this point.
-    # if no argument is provided print help and quit
-    if len(argv) == 0:
-        parser.print_help()
-        sys.exit()
     # parse arguments in case they are provided.
     x=parser.parse_args()
     active=x.active
     quiet=x.quiet
-    user_input=x.target
+    single_input=x.single_target
+    file_input=x.file_target
     output=x.output
+    # if no target is provided print help and quit.
+    if not (single_input or file_input):
```

Comment:
So whil you are at it. One more option that i always wanted to add. If there is no option switch use say someone executed datasploit in following manner.
./datasploit.py dmain.tld
./datasploit.py file_with_targets

Should we handle such situations check if the input is a file or a text and if file use that file as input otherwise if text use that as text input etc

Key is no -i or -f option is used so arg parse will not kick in.

What do you suggest.

Reply:
I initially thought about designing this patch in a _smart_ fashion that would auto-detect if file or if string input as you describe. However, the more I brainstormed it the more areas of conflict I was able to think of, and the need to write in for potential edge cases grew to a point I thought it way too much code and error checking (while still not being able to account for nearly all cases). For me, that it just wasn't the cleanest way to go.

A quick example would be - lets say someone wanted to research the domain `datasploit.py` having just "installed" datasploit via `git clone`. It would recognize the file, and explode on reading the input line by line. We could write in an exception to check for that, or to check for `#!`'s, and then what about python (or other code/etc.) files that don't use shebangs, like `osint_runner.py` (which would be an odd domain to own, for certain). Okay, we'll we could check if the first lines are `imports`. Okay, but then what about files that don't call any imports. And that's just focusing on files that may be code.

In the end, I decided it was smarter to force the user to be conscious and just use args to specify file/string input.

Reply:
fully in agreement. This was something that came in my mind and i kind of drew simmilar conclusions, its good to get assurances from someone else that i am not crazy and my logic did make sense sometimes. Thanks for confirming that.